internal/resource/state_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package resource_test
```

Comment:
todo: Update to resource package - we're not putting tests in a separate package unless absolutely necessary.